src/main/scala/org/http4s/finagle/Finagle.scala

```diff
@@ -107,18 +108,18 @@ object Finagle {
       body: Stream[F, Byte],
       writer: Writer[Buf]): Stream[F, Unit] = {
     import com.twitter.finagle.http.Chunk
-    (body.chunks.map(a => Chunk.fromByteArray(a.toArray).content).evalMap { a =>
+    body.chunks.map(a => Chunk.fromByteArray(a.toArray).content).evalMap { a =>
       toF(writer.write(a))
-    }) ++ Stream.eval { toF(writer.close()) }
+    } ++ Stream.eval { toF(writer.close()) }
   }
 
   private def toStream[F[_]](buf: Buf): Stream[F, Byte] =
     Stream.chunk[F, Byte](Chunk.array(Buf.ByteArray.Owned.extract(buf)))
 
-  def toHttp4sResp[F[_]](resp: Resp): Response[F] =
+  private def toHttp4sResp[F[_]](resp: Resp): Response[F] =
     Response[F](
-      status = Status(resp.status.code)
-    ).withHeaders(Headers(resp.headerMap.toList.map { case (name, value) => Header(name, value) }))
+      status = Status.fromInt(resp.status.code).getOrElse(Status.InternalServerError)
```

Comment:
I'm not sure about this part.
Using `Status` constructor was deprecated, `fromInt` returns `ParseResult` which is `Either` under the hood.
What would be the best behaviour here, if for whatever reason parsing fails?

Reply:
If the code is not a valid http code which is probably more fatal than 500
I think another way is to change the method type to `toHttp4sResp[F[_]](resp: Resp): F[Response[F]]` to indicate it is not pure
so we can flatMap this method in line 45 instead of map
convert either to F can be similar to line 124 `toF`

Reply:
I've posted an update shortly after.
I used `toHttpResponse` from `ParseFailure` to just respond with the failed message and will keep the method pure.